src/node_lib.h

```diff
@@ -9,93 +9,148 @@
+     * 
+     * Stops the Node.js engine.
+     * This is done in two steps:
+     * 1. Issues the Node.js event loop to no longer accept any incoming events.
+     * 2. Waits for the event loop to be empty and then executes clean up code.
+     */
     NODE_EXTERN int Deinitialize();
 
-    /*
-    Executes a given JavaScript file and returns once the execution has finished.
-    *Important*: Node.js has to have been initialized by calling Initialize().
-    */
+    /**
+     * @brief Executes the content of a given JavaScript file.
+     * 
+     * Loads and executes the content of the given file.
+     * This method returns after the script was evaluated once.
+     * This means, that any pending events will not be processes as long as 
+     * `ProcessEvents` or `RunEventLoop` is not called.
+     * @param path The path to the JavaScript file.
+     * @return The return value of the given JavaScript file.
+     */
     NODE_EXTERN v8::Local<v8::Value> Run(const std::string & path);
 
     /*********************************************************
      * Handle JavaScript events
      *********************************************************/
 
-    /*
-    Processes the pending event queue of a *running* Node.js engine once.
-    */
+    /**
+     * @brief Executes the Node.js event loop once.
+     * 
+     * Processes all currently pending events in the Node.js event loop.
+     * This method returns immediately if there are no pending events.
+     * @return True, if more events need to be processed. False otherwise.
+     */
     NODE_EXTERN bool ProcessEvents();
 
-    /*
-    Starts the execution of the Node.js event loop, which processes any events in JavaScript.
-    Additionally, the given callback will be executed once per main loop run.
-    *Important*: Call `Initialize()` before using this method.
-    */
+    /**
+     * @brief Starts the execution of the Node.js event loop. Calling the given callback once per loop tick.
+     * 
+     * Executes the Node.js event loop as long as events keep coming.
+     * Once per loop execution, after events were processed, the given callback is executed.
+     * The event loop can be paused by calling `StopEventLoop`.
+     * @param callback The callback, which should be executed periodically while the calling thread is blocked.
+     */
     NODE_EXTERN void RunEventLoop(const std::function<void()> & callback);
 
 
     /*********************************************************
      * Stop Node.js engine
      *********************************************************/
 
-    /*
-    Stops the Node.js event loop after its current execution. Execution can be resumed by calling RunEventLoop() again.
-    */
+    /**
+     * @brief Issues the Node.js event loop to stop.
+     * 
+     * Issues the Node.js event loop to stop.
+     * The event loop will finish its current execution.
+     * This means, that the loop is not stopped when this method returns.

```

Comment:
Suggestion for more clarity: "...that the loop is not guaranteed to have stopped when this method returns."